examples/computer_vision/unets_tf_keras/model_def.py

```diff
@@ -61,7 +61,8 @@ def download_weights(self):
 
         # Use a file lock so only one worker on each node does the download
         with filelock.FileLock(os.path.join(weights_dir, "download.lock")):
-            urllib.request.urlretrieve(mobilenet_link, weights_dir + data_file)
+            if not os.path.exists(weights_dir + data_file):
+                urllib.request.urlretrieve(mobilenet_link, weights_dir + data_file)
```

Comment:
What do you think about doing some sort of protection to ensure we recover well if the download fails halfway through?  Maybe just use a `try/except` block to delete the file if `urlretrieve()` fails?

Or does `urlretrieve` ensure that partial downloads are deleted?  It doesn't document any behavior like that.

Reply:
I think if `urlretrieve` raises an exception we are in bad shape, anyway.

Reply:
I guess I'm thinking of cases where every execution is not done in a clean docker filesystem.  If we had local training mode, or if the user bind-mounted  /tmp/data into the container to save time on consecutive trial executions, then we would want subsequent runs to be able to complete successfully in spite of the initial failure.

Reply:
as we discussed offline, rather than catching the exception and deleting the file, I am downloading into a temporary file, then copying to desired destination.